core/node/events/miniblock.go

```diff
@@ -171,56 +175,163 @@ func NewMiniblockInfoFromBytes(bytes []byte, expectedBlockNumber int64) (*Minibl
 			Func("NewMiniblockInfoFromBytes")
 	}
 
-	return NewMiniblockInfoFromProto(&pb, NewMiniblockInfoFromProtoOpts{ExpectedBlockNumber: expectedBlockNumber})
+	return NewMiniblockInfoFromProto(&pb, opts)
 }
 
-func NewMiniblockInfoFromBytesWithOpts(bytes []byte, opts NewMiniblockInfoFromProtoOpts) (*MiniblockInfo, error) {
+func NewMiniblockInfoFromBytes(bytes []byte, expectedBlockNumber int64) (*MiniblockInfo, error) {
 	var pb Miniblock
 	err := proto.Unmarshal(bytes, &pb)
 	if err != nil {
 		return nil, AsRiverError(err, Err_INVALID_ARGUMENT).
 			Message("Failed to decode miniblock from bytes").
 			Func("NewMiniblockInfoFromBytes")
 	}
+	expected := (*int64)(nil)
+	if expectedBlockNumber > -1 {
+		expected = &expectedBlockNumber
+	}
+	return NewMiniblockInfoFromProto(&pb, NewMiniblockInfoFromProtoOpts{ExpectedBlockNumber: expected})
+}
+
+func NewMiniblockInfoFromBytesWithOpts(bytes []byte, opts NewMiniblockInfoFromProtoOpts) (*MiniblockInfo, error) {
+	var pb Miniblock
+	err := proto.Unmarshal(bytes, &pb)
+	if err != nil {
+		return nil, AsRiverError(err, Err_INVALID_ARGUMENT).
+			Message("Failed to decode miniblock from bytes").
+			Func("NewMiniblockInfoFromBytesWithOpts")
+	}
 
 	return NewMiniblockInfoFromProto(&pb, opts)
 }
 
 type NewMiniblockInfoFromProtoOpts struct {
-	ExpectedBlockNumber int64
-	DontParseEvents     bool
+	ExpectedBlockNumber               *int64
+	ExpectedLastMiniblockHash         *common.Hash
```

Comment:
I think it should be
```suggestion
	ExpectedPrevMiniblockHash         *common.Hash
```